geps/gep-2648/index.md

```diff
@@ -0,0 +1,415 @@
+//
+// Note that a Target of a Policy is also a valid Ancestor, so for objects where
+// the Target is the relevant object for status, this struct SHOULD still be used.
+type PolicyAncestorStatus struct {
+	// AncestorRef corresponds with a ParentRef in the spec that this
+	// RouteParentStatus struct describes the status of.
+	AncestorRef ParentReference `json:"ancestorRef"`
+
+	// ControllerName is a domain/path string that indicates the name of the
+	// controller that wrote this status. This corresponds with the
+	// controllerName field on GatewayClass.
+	//
+	// Example: "example.net/gateway-controller".
+	//
+	// The format of this field is DOMAIN "/" PATH, where DOMAIN and PATH are
+	// valid Kubernetes names
+	// (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names).
+	//
+	// Controllers MUST populate this field when writing status. Controllers should ensure that
+	// entries to status populated with their ControllerName are cleaned up when they are no
+	// longer necessary.
+	ControllerName GatewayController `json:"controllerName"`
+
+	// Conditions describes the status of the Policy with respect to the given Ancestor.
+	//
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=8
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+
+// PolicyStatus defines the common attributes that all Policies SHOULD include
+// within their status.
+type PolicyStatus struct {
+	// Ancestors is a list of ancestor resources (usually Gateways) that are
+	// associated with the route, and the status of the route with respect to
+	// each ancestor. When this route attaches to a parent, the controller that
+	// manages the parent and the ancestors MUST add an entry to this list when
+	// the controller first sees the route and SHOULD update the entry as
+	// appropriate when the relevant ancestor is modified.
+	//
+	// Note that choosing the relevant ancestor is left to the Policy designers;
+	// an important part of Policy design is designing the right object level at
+	// which to namespace this status.
+	//
+	// Note also that implementations MUST ONLY populate ancestor status for 
+	// the Ancestor resources they are responsible for. Implementations MUST
+	// use the ControllerName field to uniquely identify the entries in this list
+	// that they are responsible for.
+	//
+	// A maximum of 32 ancestors will be represented in this list. An empty list
+	// means the Policy is not relevant for any ancestors.
+	//
+	// +kubebuilder:validation:MaxItems=32
+	Ancestors []PolicyAncestorStatus `json:"ancestors"`
+}
+```
+
+## Examples
+
+### Hypothetical TLSMinimumVersionPolicy
+
+The following hypothetical Policy sets the minimum TLS version required on a
+Gateway Listener:
+
+```yaml
+apiVersion: networking.example.io/v1alpha1
+kind: TLSMinimumVersionPolicy
+metadata:
+  name: minimum12
+  namespace: appns
```

Comment:
Is this missing the label?
```suggestion
  namespace: appns
  label: `gateway.networking.k8s.io/policy:direct'
```

Reply:
Yes, it's from an earlier version that didn't have the requirement, thanks.